llm-tracing/main.py

```diff
@@ -0,0 +1,90 @@
+from fastapi import FastAPI
+from traceloop.sdk import Traceloop
+from dotenv import load_dotenv
+import os
+from pydantic import BaseModel
+from llama_index.core.agent.workflow import FunctionAgent
+from llama_index.llms.openai import OpenAI
+from opentelemetry.instrumentation.openai import OpenAIInstrumentor
+
+OpenAIInstrumentor().instrument()
+
+load_dotenv()
+
+# Ensure APP_NAME environment variable exists
+if not os.getenv("APP_NAME"):
+    raise ValueError("APP_NAME environment variable is not set")
+
+Traceloop.init(disable_batch=True, 
+               app_name=os.getenv("APP_NAME"))
+
+client = OpenAI()
+
+app = FastAPI()
+
+def multiply(a: float, b: float) -> float:
+    """Useful for multiplying two numbers."""
+    try:
+        return a * b
+    except TypeError:
+        raise TypeError("Arguments must be numbers")
+
+class OpenAIRequest(BaseModel):
+    model_name: str = "gpt-4"  # Example: "gpt-4", "gpt-3.5-turbo"
+    user_message: str = "What is 5 times 7?"  # Example user query for multiplication
+
+async def generate_openai_response(client, model_name, user_message):
+    try:
```

Comment:
This try catch block is not required as you already have on in the caller. `generate_response`/